File: Java/ProtoDebugger/src/protodebugger/model/descriptors/TextFieldDescriptorContainer.java
<<<<<<< HEAD
package protodebugger.model.descriptors;

import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.Text;
import org.eclipse.swt.widgets.Widget;

import com.google.protobuf.Descriptors;
import com.google.protobuf.GeneratedMessage.Builder;


public class TextFieldDescriptorContainer extends FieldDescriptorContainer {
	private Text textField;
	
	public TextFieldDescriptorContainer(Descriptors.FieldDescriptor field)
	{
		super(field);
	}
	@Override
	public Object getValue() {
		if(value != null)
			return value.toString();
		else
			return "";
	}

	@Override
	public boolean buildMsg(Builder<?> build) {
		if(field.isOptional() && textField.getText().equals(""))
			return false;
		if(field.isRepeated())
			build.addRepeatedField(field, textField.getText());
		else
			build.setField(field, textField.getText());
		return true;
			
	}

	@Override
	public String toString() {
		return "TextField name = '" + name + "'";
	}

	@Override
	public Widget getWidget(Composite parent) {
		if(textField == null)
		{
			textField = new Text(parent, SWT.BORDER);
			textField.setText((String)getValue());
		}
		else if(textField.getParent() != parent)
		{
			textField.setParent(parent);
		}
		return textField;
		
	}

	@Override
	public Composite getParent() {
		if(textField != null)
			return textField.getParent();
		return null;
	}

	
}
=======
package protodebugger.model.descriptors;

import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.Text;
import org.eclipse.swt.widgets.Widget;

import com.google.protobuf.Descriptors;
import com.google.protobuf.GeneratedMessage.Builder;


public class TextFieldDescriptorContainer extends FieldDescriptorContainer {
	private Text textField;
	
	public TextFieldDescriptorContainer(Descriptors.FieldDescriptor field)
	{
		super(field);
	}
	@Override
	public Object getValue() {
		if(value != null)
			return value.toString();
		else if(defaultValue != null)
			return defaultValue.toString();
		else
			return "";
	}

	@Override
	public boolean buildMsg(Builder<?> build) {
		if(field.isOptional() && textField.getText().equals(""))
			return false;
		if(field.isRepeated())
			build.addRepeatedField(field, textField.getText());
		else
			build.setField(field, textField.getText());
		return true;
			
	}

	@Override
	public void setValue(Object value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return "TextField name = '" + name + "'";
	}

	@Override
	public Widget getWidget(Composite parent) {
		if(textField == null)
		{
			textField = new Text(parent, SWT.BORDER);
			textField.setText((String)getValue());
		}
		else if(textField.getParent() != parent)
		{
			textField.setParent(parent);
		}
		return textField;
		
	}

	@Override
	public Composite getParent() {
		if(textField != null)
			return textField.getParent();
		return null;
	}

}
>>>>>>> 5a14f9bb2eb2e5da316d0185feeade7969c536bd

File: CPP/ProtoDebugger/src/protodebugger/model/descriptors/TextFieldDescriptorContainer.cpp
#include "TextFieldDescriptorContainer.h"
#include <sstream>

using namespace std;
using namespace google;
using namespace protobuf;

QWidget * TextFieldDescriptorContainer::getWidget(QWidget * parent)
{
    stringstream ss;
    if(m_textField == NULL)
    {
        m_textField = new QLineEdit(parent);

        ss << m_value;
        m_textField->setText(QString::fromStdString(ss.str()));
    }
    else if (m_textField->parentWidget() != parent)
    {
        m_textField->setParent(parent);
    }

    return m_textField;
}

QWidget * TextFieldDescriptorContainer::getParent()
{
    if(m_textField != NULL)
    {
       return m_textField->parentWidget();
    }

    return NULL;
}

Object * TextFieldDescriptorContainer::getValue()
{
    stringstream ss;
    if( m_value != NULL)
    {
        ss << m_value;
        return (Object *) ss.str().c_str();
    }
    else if (m_defaultValue != NULL)
    {
        ss << m_defaultValue;
        return (Object *) ss.str().c_str();
    }
    return (Object *) "";
}

void TextFieldDescriptorContainer::setValue(Object * value)
{
    m_value = value;
}

QString TextFieldDescriptorContainer::toString()
{
    string str = "TextField name = " + this->m_name;
    return QString(str.c_str());
}

bool TextFieldDescriptorContainer::buildMsg(Message *msg)
{
    if(m_field->is_optional() && m_textField->text().compare("") == 0)
    {
        return false;
    }
    if (m_type == FieldDescriptor::CPPTYPE_STRING)
    {
        if(m_field->is_repeated())
        {
             msg->GetReflection()->AddString(msg, m_field, m_textField->text().toStdString());
        }
        else
        {
            msg->GetReflection()->SetString(msg, m_field, m_textField->text().toStdString());
        }
    }
    return true;
}

